src/modules/cmdpal/Microsoft.CmdPal.UI/Controls/SearchBar.xaml.cs

```diff
@@ -44,16 +46,21 @@ private static void OnCurrentPageViewModelChanged(DependencyObject d, Dependency
         if (d is SearchBar @this
             && e.NewValue is PageViewModel page)
         {
-            // TODO: In some cases we probably want commands to clear a filter somewhere in the process, so we need to figure out when that is.
+            // TODO: In some cases we probably want commands to clear a filter
+            // somewhere in the process, so we need to figure out when that is.
             @this.FilterBox.Text = page.Filter;
+            @this.FilterBox.Select(@this.FilterBox.Text.Length, 0);
         }
     }
 
     public SearchBar()
     {
         this.InitializeComponent();
+        WeakReferenceMessenger.Default.Register<GoHomeMessage>(this);
     }
 
+    public void ClearSearch() => this.FilterBox.Text = string.Empty;
```

Comment:
This doesn't have to be public anymore, eh? Should we just move the line into the Receive body for GoHome?